### ISSUE TITLE
common - podOptions hostUsers not being set

### ISSUE THREAD

**Issue by Kampe** [truecharts/library-charts#823]
### chart Name

common

### Operating System

Talos-OS 1.7
Kubernetes 1.30

### Deployment Method

Helm

### Chart Version

latest

### Kubernetes Events

```Shell
n/a
```

### chartlication Logs

```Shell
N/A
```

### Chart Configuration

podOptions:
  hostUsers: false

### Describe the bug

podOptions:
  hostUsers: false
  
has no effect for any charts that utilize common.

### To Reproduce

install any true chart and set podOptions.hostUsers 

### Expected Behavior

expected that the hostUsers field is set downstream on pods. 

### Screenshots

n/a

### Additional Context

n/a

### I've read and agree with the following

- [X] I've checked all open and closed issues and my issue is not there.
- [X] I've understand TrueNAS SCALE Apps are deprecated and TrueCharts does not accept TrueNAS SCALE related issues
- [X] I've prefixed my issue title with `[Chart-Name]`

**Comment by stavros-k** [comment]
Currently the way we have it like this:

- Defaults to `false`
- If it matches one of the conditions we set, we set it to `true`
- If podOptions.hostUsers is `true`, we set it to true regardless of the calculations above
- If $wokrload.podSpec.hostUsers is `true`, we set it to true regardless of the calculations above

Currently there is no way to set it to `false` if one of the conditions is matched.

TL;DR; currently users can manually only opt-in to hostUsers if the conditions are not met.
There is no opt-out.

**Comment by Ornias1993** [comment]
Yeah so this is expected behavior...

**Comment by Kampe** [comment]
This will become increasingly more of a problem if its not set as of Kubernetes 1.30 as it becomes a required field and is set automatically.  This causes issues in any gitops tools.

**Comment by stavros-k** [comment]
Functionality will change to this:

If the key is missing, or is not a bool, we will set it based on our predefined conditions.
If the key is set and its a bool. We will use that value.
